020-quarkus-http-non-application-endpoints/src/test/java/io/quarkus/qe/non_application/endpoint/BackwardCompatibilityTest.java

```diff
@@ -1,37 +1,35 @@
 package io.quarkus.qe.non_application.endpoint;
 
+import static io.quarkus.qe.non_application.endpoint.CommonNonAppEndpoint.QUARKUS_PROFILE;
+import static io.quarkus.qe.non_application.endpoint.CommonNonAppEndpoint.nonAppEndpoints;
 import static io.restassured.RestAssured.given;
 import static org.hamcrest.Matchers.endsWith;
 import static org.hamcrest.Matchers.in;
 
+
 import java.util.Collections;
 
 import org.jboss.shrinkwrap.api.ShrinkWrap;
 import org.jboss.shrinkwrap.api.spec.JavaArchive;
-import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.RegisterExtension;
 
 import io.quarkus.qe.http.non_application.endpoint.HelloResource;
 import io.quarkus.test.QuarkusProdModeTest;
 
-public class BackwardCompatibilityTest extends CommonNonAppEndpoint {
-    private static final String BASE_PATH = "/q";
+public class BackwardCompatibilityTest {
+    protected static final String BASE_PATH = "/q";
 
     @RegisterExtension
     static final QuarkusProdModeTest backwardScenario = new QuarkusProdModeTest()
+            .setBuildNative(Boolean.getBoolean(System.getProperty(QUARKUS_PROFILE)))
```

Comment:
I know It's working because on surefire, we don't add the quarkus.profile property, but it should return true if and only if the quarkus.profile property is "native". Now, it returns true if the property is present.